mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/IGG-cf_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/IGG-cf_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/IGG-cf_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/IGG-cf_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/IGG-cf_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf10_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf10_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf10_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf10_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf10_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf10_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf10_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf10_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf10_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf10_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf10_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf10_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf10_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf10_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf10_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf10_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf10_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf10_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf10_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf10_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf10_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf11_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf11_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf11_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf11_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf11_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf11_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf11_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf11_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf11_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf11_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf11_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf11_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf11_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf11_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf11_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf11_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf11_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf11_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf11_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf11_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf11_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/QCF1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/QCF1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/QCF1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/QCF1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/QCF1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/QCF1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/QCF1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/QCF1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/QCF1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/QCF1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/QCF1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/QCF1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/QCF1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/QCF1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/QCF1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/QCF1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/QCF1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/QCF1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/QCF1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/QCF1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF1_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/QCF2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/QCF2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/QCF2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/QCF2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/QCF2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/QCF2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/QCF2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/QCF2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/QCF2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/QCF2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/QCF2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/QCF2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/QCF2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/QCF2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/QCF2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/QCF2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/QCF2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/QCF2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/QCF2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/QCF2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/QCF2_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf4_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf4_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf4_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf4_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf4_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf4_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf4_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf4_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf4_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf4_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf4_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf4_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf4_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf4_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf4_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf4_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf4_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf4_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf4_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf4_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf4_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf5_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf5_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf5_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf5_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf5_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf5_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf5_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf5_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf5_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf5_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf5_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf5_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf5_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf5_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf5_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf5_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf5_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf5_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf5_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf5_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf5_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf6_SS_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf6_SS_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf6_SS_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf6_SS_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf6_SS_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf6_SS_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf6_SS_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf6_SS_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf6_SS_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf6_SS_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf6_SS_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf6_SS_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf6_SS_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf6_SS_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf6_SS_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf6_SS_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf6_SS_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf6_SS_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf6_SS_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf6_SS_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf6_SS_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf7_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf7_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf7_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf7_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf7_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf7_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf7_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf7_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf7_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf7_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf7_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf7_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf7_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf7_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf7_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf7_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf7_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf7_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf7_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf7_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf7_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf8_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf8_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf8_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf8_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf8_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf8_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf8_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf8_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf8_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf8_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf8_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf8_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf8_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf8_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf8_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf8_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf8_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf8_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf8_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf8_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf8_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf9_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf9_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf9_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf9_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf9_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf9_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf9_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf9_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf9_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf9_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf9_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf9_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf9_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf9_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf9_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf9_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf9_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf9_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf9_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf9_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf9_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All1_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All2_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_All3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_All3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_All3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_All3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_All3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_All3_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_DE1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_DE1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_DE1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_DE1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_DE1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_DE1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_DE1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_DE1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_DE1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_DE1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_DE1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_DE1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_DE1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_DE1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_DE1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_DE1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_DE1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_DE1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_DE1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_DE1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_DE1_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_ED1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_ED1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_ED1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_ED1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_ED1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_ED1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_ED1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_ED1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_ED1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_ED1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_ED1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_ED1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_ED1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_ED1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_ED1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_ED1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_ED1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_ED1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_ED1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_ED1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_ED1_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Exo1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Exo1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Exo1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Exo1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Exo1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Exo1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Exo1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Exo1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Exo1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Exo1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Exo1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Exo1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Exo1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Exo1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Exo1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Exo1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Exo1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Exo1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Exo1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Exo1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Exo1_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag1_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag2_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Frag3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Frag3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Frag3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Frag3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Frag3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Frag3_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS1_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS2_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_HS3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_HS3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_HS3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_HS3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_HS3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_HS3_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_L1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_L1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_L1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_L1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_L1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_L1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_L1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_L1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_L1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_L1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_L1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_L1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_L1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_L1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_L1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_L1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_L1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_L1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_L1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_L1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L1_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_L2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_L2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_L2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_L2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_L2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_L2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_L2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_L2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_L2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_L2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_L2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_L2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_L2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_L2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_L2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_L2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_L2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_L2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_L2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_L2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_L2_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE01_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE01_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE01_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE01_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE01_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE01_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE01_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE01_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE01_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE01_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE01_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE01_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE01_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE01_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE01_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE01_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE01_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE01_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE01_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE01_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE01_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE100_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE100_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE100_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE100_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE100_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE100_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE100_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE100_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE100_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE100_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE100_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE100_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE100_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE100_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE100_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE100_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE100_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE100_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE100_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE100_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE100_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE1K_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE1K_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE1K_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE1K_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE1K_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE1K_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE1K_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE1K_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE1K_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE1K_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE1K_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE1K_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE1K_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE1K_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE1K_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE1K_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE1K_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE1K_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE1K_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE1K_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1K_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_LE1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_LE1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_LE1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_LE1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_LE1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_LE1_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Na1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Na1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Na1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Na1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Na1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Na1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Na1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Na1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Na1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Na1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Na1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Na1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Na1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Na1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Na1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Na1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Na1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Na1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Na1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Na1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Na1_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA2_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA2_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA2_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA2_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA2_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA3_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA3_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA3_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA3_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA3_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA4_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA4_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA4_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_NA4_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA4_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA4_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA4_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_NA4_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA4_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA4_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA4_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_NA4_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA4_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA4_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA4_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_NA4_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA4_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA4_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA4_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_NA4_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_NA4_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Phos1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Phos1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Phos1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/genes.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/primary_no_sncRNA_tRNA_rRNA_repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/counts/Qcf_Phos1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.primary_no_sncRNA_tRNA_rRNA_repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.sncRNA.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/sncRNA.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.sncRNA.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/sncRNA.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/sncRNA.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Phos1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/sncRNA.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/sncRNA.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Phos1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/sncRNA.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Phos1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/sncRNA.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/sncRNA_x_protein.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/sncRNA.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/sncRNA/Qcf_Phos1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.sncRNA.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/smallRNA/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/smallRNA/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/smallRNA/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Phos1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/smallRNA/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/smallRNA/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Phos1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/smallRNA/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Phos1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/smallRNA/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/smallRNA.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/smallRNA/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                       |awk  '{print $2,$3,$4,$5,$6,$7, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/smallRNA/Qcf_Phos1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.aligned.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/rRNA_mt/aligned.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.aligned.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/rRNA_mt/aligned.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/rRNA_mt/aligned.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Phos1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/rRNA_mt/aligned.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/rRNA_mt/aligned.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Phos1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/rRNA_mt/aligned.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Phos1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/rRNA_mt/aligned.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0    -t 0 --ct 6|  poisson_umi_adjustment.py -i - -o - --umi 6                      | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/new_genes/rRNA_mt.bed -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/rRNA_mt/aligned.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/rRNA_mt/Qcf_Phos1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.aligned.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Phos1_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Phos1_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Phos1_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/Qcf_Phos1_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Qcf_Phos1_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/S96-cfD_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/S96-cfD_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/S96-cfD_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/S96-cfD_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cfD_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/S96-cf_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/S96-cf_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/S96-cf_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/S96-cf_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/S96-cf_R1_001.repeats.bam_TEMP
mkdir -p /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001.repeats.bam_TEMP ; cat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001/Combined/repeats.bam | samtools view -bF 4 -F 256 -F 2048   | bam_to_bed.py -i - --add_cigar --primary             | sort -k1,1 -k2,2n -k3,3n -k6,6 --temporary-directory=/stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001.repeats.bam_TEMP             | bgzip             > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001/Combined/repeats.bed.gz ; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001/Combined/repeats.all.sense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/try_R1_001.all.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001/Combined/repeats.bed.gz                                             | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001/Combined/repeats.all.antisense.intersected.bed.gz                      | zcat                      | awk '{print $8,$9,$10,$11,$12,$13,$14,$15}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/try_R1_001.all.antisense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -s  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001/Combined/repeats.dedup.sense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/try_R1_001.dedup.sense.counts; zcat /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001/Combined/repeats.bed.gz                      | deduplicate_bed.py -i - -d '_' -f 0                         | bedtools intersect -a - -b /stor/work/Lambowitz/ref/hg19/genome/rmsk.bed.gz -f 0.1  -S  -wao                     | bgzip                     | tee /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001/Combined/repeats.dedup.antisense.intersected.bed.gz                      | zcat                      | awk '{print $7,$8,$9,$10,$11,$12,$13,$14}' OFS='\t'                     | sort                     | uniq -c                      | awk  '{print $2,$3,$4,$5,$6,$7,$8, $9, $1}'   OFS='\t'                       > /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/Counts/all_counts/repeats/try_R1_001.dedup.antisense.counts; rm -rf /stor/work/Lambowitz/cdw2854/cfNA/tgirt_map/try_R1_001.repeats.bam_TEMP
